File: express-backend/src/models/Alert.ts
import mongoose, { Schema, type Document, type Model } from 'mongoose';

export interface AlertDocument extends Document {
  patient_id?: mongoose.Types.ObjectId;
  severity: string;
  type: string;
  title: string;
  body?: string;
  acknowledged_at?: Date | null;
  acknowledged_by?: mongoose.Types.ObjectId | null;
  created_at: Date;
}

const AlertSchema = new Schema<AlertDocument>(
  {
    patient_id: { type: Schema.Types.ObjectId, ref: 'Patient' },
    severity: { type: String, required: true },
    type: { type: String, required: true },
    title: { type: String, required: true },
    body: { type: String },
    acknowledged_at: { type: Date, default: null },
    acknowledged_by: { type: Schema.Types.ObjectId, ref: 'User', default: null },
    created_at: { type: Date, default: () => new Date() },
  },
  {
    collection: 'alerts',
  },
);

export const Alert: Model<AlertDocument> =
  mongoose.models.Alert || mongoose.model<AlertDocument>('Alert', AlertSchema);

File: backend/app/schemas/alert.py
"""Alert (clinical/system) document schema."""
from datetime import datetime, timezone
from typing import Optional

from pydantic import BaseModel, Field

from .enums import AlertSeverity
from .objectid import PyObjectId


def _utc_now() -> datetime:
    return datetime.now(timezone.utc)


class AlertDoc(BaseModel):
    """Mongo document shape for alerts collection. patient_id optional for system-wide alerts."""

    id: Optional[PyObjectId] = Field(None, alias="_id")
    patient_id: Optional[PyObjectId] = None
    severity: AlertSeverity
    type: str
    title: str
    body: Optional[str] = None
    acknowledged_at: Optional[datetime] = None
    acknowledged_by: Optional[PyObjectId] = None
    created_at: datetime = Field(default_factory=_utc_now)

    model_config = {"populate_by_name": True, "arbitrary_types_allowed": True}
